### PR TITLE
Store and send CSRF tokens to the API

### PR DESCRIPTION
**Pull request by kumar303**
Part of https://github.com/mozilla/payments-service/issues/46

I can't think of an easy way to test this. Any ideas?

Going forward, I think we could make an API abstraction (perhaps around fetch, see https://github.com/mozilla/payments-ui/issues/84). If we design the abstraction with testability in mind we could stub it out in tests so that components are fed mock return data. This would allow the tests to make assertions about what components do with the data.

### CODE REVIEW

File: public/jsx/views/login.jsx [hunk 1]
```diff
@@ -29,9 +29,17 @@ module.exports = React.createClass({
       method: 'post',
       url: '/api/auth/sign-in/',
       context: this,
-    }).then(function() {
+    }).then(function(data) {
       if (this.isMounted()) {
         this.setState({'is_logged_in': true}); // eslint-disable-line
+
+        console.log('setting CSRF token for subsequent requests:', data.csrf_token);
+        $.ajaxSetup({
```
Comment (muffinresearch) [inline public/jsx/views/login.jsx]: Presumably this is going to add the header and not unset the dataType setting? We're already running it here: https://github.com/mozilla/payments-ui/blob/master/public/js/main.js#L7

**Comment by muffinresearch** [comment]
I'm ok with not worrying about the tests for now give that if this isn't sent correctly the API will lock us out. We can definitely do something down the line (along the lines of what you're suggesting).

r+wc

File: public/jsx/views/login.jsx (hunk 1 above)
Comment (kumar303) [inline public/jsx/views/login.jsx]: yeah, it looks like something you can run multiple times and not override old values. I'll test it and see. It's treating jQuery like a static store of saved data. If I were to do it some other way we'd still need the same concept of keeping a static store of data in memory.

**Comment by kumar303** [comment]
depends on https://github.com/mozilla/payments-service/pull/52

**Pull request merged**